src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ILEmit/ILEmitResolverBuilder.cs

```diff
@@ -96,21 +96,22 @@ private GeneratedMethod BuildTypeNoCache(ServiceCallSite callSite)
             ILGenerator ilGenerator = dynamicMethod.GetILGenerator(512);
             ILEmitResolverBuilderRuntimeContext runtimeContext = GenerateMethodBody(callSite, ilGenerator);
 
-#if SAVE_ASSEMBLIES
+#if SAVE_ASSEMBLIES && NETFRAMEWORK
```

Comment:
```suggestion
#if SAVE_ASSEMBLIES
```

The library's `.csproj` file already allows defining `SAVE_ASSEMBLIES` only on .NET Framework.